core/storage/base_model/gae_models.py

```diff
@@ -193,11 +197,12 @@ def get_new_id(cls, entity_name):
                 of attempts.
         """
         try:
-            entity_name = unicode(entity_name).encode(encoding='utf-8')
+            entity_name = python_utils.convert_to_bytes(entity_name).encode(
```

Comment:
Encoding again is redundant, just return the bytes.

Reply:
Done